packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts

```diff
@@ -1,29 +1,25 @@
 import { GraphQLObjectType } from 'graphql';
-import { NoDataError, FieldTransform, TransformType } from '@graphback/runtime';
+import { NoDataError, TransformType, GraphbackOrderBy, GraphbackPage } from '@graphback/runtime';
 import { getDatabaseArguments } from '@graphback/core';
 import { ObjectId } from 'mongodb';
 import { MongoDBDataProvider } from '@graphback/runtime-mongo';
 
 /**
- * Mongo provider that contains special handlers for offix conflict resolution format:
- *
- * https://offix.dev/docs/conflict-server#structure-of-the-conflict-error
+ * Mongo provider that attains data synchronization using soft deletes
  */
-export class DatasyncMongoDBDataProvider<Type = any, GraphbackContext = any> extends MongoDBDataProvider<Type, GraphbackContext> {
+export class DataSyncMongoDBDataProvider<Type = any, GraphbackContext = any> extends MongoDBDataProvider<Type, GraphbackContext> {
 
   public constructor(baseType: GraphQLObjectType, client: any) {
     super(baseType, client);
-  }
-
-  public async create(data: any): Promise<Type> {
-    if (!data.version) {
-      data.version = 1;
-    }
 
-    return super.create(data);
+    // Mark any created items with 'deleted' flag set to false
+    this.fieldTransformMap[TransformType.CREATE].push({
```

Comment:
I think we overusing this. Since we will have provider for every strategy we might directly out version,_deleted etc. into create method. 
No strong opinion but at the first glance looks like overkill without any functional reason. is there functional benefit to have it this way?

Reply:
> I think we overusing this. Since we will have provider for every strategy we might directly out version,_deleted etc. into create method.

I have removed version field, do we need it?

> No strong opinion but at the first glance looks like overkill without any functional reason. is there functional benefit to have it this way?

Not really, was just trying to reuse that code

Reply:
Made some changes, please check